src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -1,13 +1,92 @@
 import React from 'react';
-import {getFollowing} from '../../../service/profile';
+import {Context} from '../../../context';
+import DiscoveryAction from '../../../context/actions/discoveryAction';
+import following from '../../../context/actions/following';
 import {getFollowedDomain} from '../../../service/domain';
+import {getFollowing} from '../../../service/profile';
 import {getFollowingTopic} from '../../../service/topics';
-import following from '../../../context/actions/following';
-import {Context} from '../../../context';
 
 const useDiscovery = () => {
   const [, followingDispatch] = React.useContext(Context).following;
   const [refreshing, setRefreshing] = React.useState(false);
+  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
+
+  const users = React.useMemo(() => {
+    return discovery.initialUsers.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `users` memoization logic seems to be assuming that if `item.following` is undefined, then the user is followed if `item.user_id_follower` is not null. This might not always be correct and could lead to incorrect follow status being displayed. It would be better to ensure that the backend always sends a valid `following` status for each user.

```diff
- following: item.following !== undefined ? item.following : item.user_id_follower !== null
+ following: item.following
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->